compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1787,8 +1793,35 @@ impl<'a> Parser<'a> {
                 Ok(true) => {}
                 Ok(false) => unreachable!(),
                 Err(mut err) => {
+                    // Qualifier keywords ordering check
+
+                    // This will allow the machine fix to directly place the keyword in the correct place
+                    let current_qual_sp = if self.check_keyword(kw::Const) {
```

Comment:
If I'm reading this correctly, we will trigger the diagnostic here below for code that has duplicate keywords, such as e.g. in

```rust
const async const fn
```

and I believe that the machine applicable suggestion will suggest a `const const async fn` which is invalid. 

It may not be critical to fix this now in this PR, but could you please at least add a test for that case?

Reply:
You are right, I will add a test for it.

`pub pub fn test() {}` already behaves like that and has a `MachineApplicable` fix, I think it is because it is expected to sometimes misplace keywords but repeating them is not something that happens often so it should be okay ? I can try to fix it in this PR or open another issue+PR combo, I don't know what's best here

Reply:
I don't believe this must be fixed in this PR, but I would consider that a bug.